lib/node/nodes/aggregate-intersection.js

```diff
@@ -77,3 +77,16 @@ var queryAggregateTemplate = Node.template([
     'WHERE ST_Intersects(_cdb_analysis_source.the_geom, _cdb_analysis_target.the_geom)',
     'GROUP BY {{=it.groupByColumns}}'
 ].join('\n'));
+
+AggregateIntersection.prototype.computeRequirements = function(databaseService, limits, callback) {
+    // we estimate the maximum possible number of rows of the result
+    var product = this.source.estimatedRequirements.numberOfRows *
+                  this.target.estimatedRequirements.numberOfRows;
+    this.estimatedRequirements = {
+        numberOfRows: product
+    };
+    this.limits = {
+        maximumNumberOfRows: Node.getNodeLimit(limits, TYPE, 'maximumNumberOfRows', 1000000)
+    };
+    return callback(null);
```

Comment:
Ok, but be careful with CPU intensive tasks and consider use `process.nextTick()` if required.